### ISSUE TITLE
Misleading error message with invalid `morphology.skeletonize` method

### ISSUE THREAD

**Issue by jamesyan-git** [scikit-image/scikit-image#6661]
### Description:

When using `skeletonize` on a 2d numpy array with an invalid method parameter, an error message is shown falsely stating that there is a mismatch in the array dimension. 

I would expect the error message to highlight the incorrect method as the issue rather than incorrect array dimensionality.

Let me know if I can provide any further detail :smi

### Way to reproduce:

```Python
import numpy as np
from skimage.morphology import skeletonize

result = skeletonize(np.ones((5, 5)), method='foo')
```

### Traceback or output:

```Shell
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\james\anaconda3\envs\skan\lib\site-packages\skimage\morphology\_skeletonize.py", line 87, in skeletonize
    raise ValueError(f'skeletonize requires a 2D or 3D image as input, '
ValueError: skeletonize requires a 2D or 3D image as input, got 2D.
```

### Version information:

```Shell
3.10.8 | packaged by conda-forge | (main, Nov 24 2022, 14:07:00) [MSC v.1916 64 bit (AMD64)]
Windows-10-10.0.19044-SP0
scikit-image version: 0.19.3
numpy version: 1.23.5
```

**Comment by jni** [comment]
Thanks for the report @jamesyan-git!

This might make a good issue for a first-time contributor — we should add an if clause checking that the method is valid first, then do the dimensionality checks. (e.g. `if method not in {'lee', 'zhang'}: raise ValueError("<error message>")`)

**Comment by lagru** [comment]
I can confirm the bug. Thanks for the report @jamesyan-git, and I agree with @jni's assessment. :+1:

**Comment by arsh73552** [comment]
Hey, Thanks for marking this as a good first issue, this seems like something I can do and I'll work on it!

**Comment by lagru** [comment]
Fixed in #6805.
